ethcore/sync/src/chain/mod.rs

```diff
@@ -1121,13 +1158,13 @@ impl ChainSync {
 	}
 
 	/// Broadcast private transaction message to peers.
-	pub fn propagate_private_transaction(&mut self, io: &mut SyncIo, packet: Bytes) {
-		SyncPropagator::propagate_private_transaction(self, io, packet);
+	pub fn propagate_private_transaction(&mut self, io: &mut SyncIo, transaction_hash: H256, packet: Bytes) {
```

Comment:
Might be worth merging the two methods (and the methods they are calling into one) and just discriminate using `enum` to avoid duplication.